### PR TITLE
replayer: Output doctest xml if launched by CLion

### PR DESCRIPTION
**Pull request by florianGla**

### CODE REVIEW

File: tools/replayer/src/replayer.c [hunk 1]
```diff
@@ -364,6 +367,22 @@ static void run_file_or_dir(const char *path) {
   }
 }
 
+static void print_clion_doctest_xml() {
+  /*
+   * Output Doctest xml, which enables the CLion test framework integration
+   * to indicate the test success status.
+   */
+  fprintf(stdout,
```
Comment (fmeum) [inline tools/replayer/src/replayer.c:375]: When printing to stdout you can simply use `printf`.

File: tools/replayer/src/replayer.c [hunk 2]
```diff
@@ -364,6 +367,22 @@ static void run_file_or_dir(const char *path) {
...
+          "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+          "\n<doctest>"
```
Comment (fmeum) [inline tools/replayer/src/replayer.c:377]: Are the `\n`s significant to CLion? If not, then I would vote for leaving them out. The visual line breaks help, but the ones in the string literal will never be seen by a human.

File: tools/replayer/src/replayer.c [hunk 3]
```diff
@@ -69,6 +69,9 @@
  * be executed. */
 static const char *SEED_CORPUS_SUFFIX = "_seed_corpus";
 
+// A nonzero value indicates that replayer was launched as a Doctest target by the CLion test framework integration.
```
Comment (fmeum) [inline tools/replayer/src/replayer.c:72]: You have to use `/* ... */` for compatibility with ancient C standards.

Comment (florianGla) [reply tools/replayer/src/replayer.c:72]: That explains the failing tests.

File: tools/replayer/src/replayer.c (hunk 2 above)
Comment (florianGla) [reply tools/replayer/src/replayer.c:377]: You are right, only the last \n is significant.

**Review by fmeum — APPROVED** [review]

**Review by fmeum — APPROVED** [review]

**Pull request merged**